contracts/src/mixins/ERC4337Mixin.sol

```diff
@@ -173,7 +222,9 @@ abstract contract ERC4337Mixin is IAccount, IHarbourStore {
             computedSafeTxHash == safeTxHash,
             UnexpectedSafeTxHash(computedSafeTxHash)
         );
-        return (safeTxHash, signer, r, vs);
+        // The computed length when properly encoded is based on the data length and the number of params
+        // 4 bytes selector + 15 params each 32 bytes + 32 bytes offset of data + 32 bytes length of data + data length
+        return (safeTxHash, signer, r, vs, data.length + 17 * 32 + 4);
```

Comment:
I think default Solidity ABI encoding pads the bytes after the selector to 32 bytes.

You could do something like (if my bitbanging memory serves me right):

```solidity
uint256 paddedDataLength = (data.length + 31) & (~31);
```